keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/AllowedTimesConstraintEvaluator.kt

```diff
@@ -254,6 +256,32 @@ class AllowedTimesConstraintEvaluator(
     return false
   }
 
+  override fun generateConstraintState(
+    artifact: DeliveryArtifact,
+    version: String,
+    deliveryConfig: DeliveryConfig,
+    targetEnvironment: Environment
+  ): ConstraintState {
+    val constraint = getConstraintForEnvironment(deliveryConfig, targetEnvironment.name, supportedType.type)
+
+    val tz: String = constraint.tz ?: dynamicConfigService.getConfig(String::class.java, "default.time-zone", "America/Los_Angeles")
+
+    return ConstraintState(
+      deliveryConfigName = deliveryConfig.name,
+      environmentName = targetEnvironment.name,
+      artifactVersion = version,
+      artifactReference = artifact.reference,
+      type = CONSTRAINT_NAME,
+      status = PASS,
```

Comment:
@lorin What if I were to pass in the status I wanted? I want to avoid user confusion in the case of a race condition (like the artifact is approved at 9:59:59 and then we call the allowed times at 10:00:01 and it is now false). The change in assumption would be that the calling function should know what happened.

Reply:
@emjburns Yeah, I think passing it in would be good.

Reply:
ok @lorin I've made this change, I'd love for you to take another look.